perspective.el

```diff
@@ -968,6 +990,249 @@ perspective beginning with the given letter."
+(defun persp-state-load (file)
+  "Restore the perspective state saved in FILE.
+
+FILE defaults to the value of persp-state-default-file if it is
+set.
+
+Frames are restored, along with each frame's perspective list.
+Each perspective's buffer list and window layout are also
+restored."
+  (interactive (list
+                (read-file-name "Restore perspective state from file: "
+                                persp-state-default-file
+                                persp-state-default-file)))
+  (unless (file-exists-p file)
+    (error "File not found"))
+  (persp-mode 1)
+  ;; before hook
+  (run-hooks 'persp-state-before-load-hook)
+  ;; actually load
+  (lexical-let ((tmp-persp-name (format "%04x%04x" (random (expt 16 4)) (random (expt 16 4))))
+                (frame-count 0)
+                (state-complete (read-from-whole-string
+                                 (with-temp-buffer
+                                   (insert-file-contents file)
+                                   (buffer-string)))))
+    ;; open all files in a temporary perspective to avoid polluting "main"
+    (persp-switch tmp-persp-name)
+    (cl-loop for file in (persp--state-complete-files state-complete) do
+             (when (file-exists-p file)
+               (find-file file)))
+    ;; iterate over the frames
+    (cl-loop for frame in (persp--state-complete-frames state-complete) do
+             (incf frame-count)
+             (when (> frame-count 1)
+               (make-frame-command))
+             ;; XXX: The condition in binding frame-persp-table and
+             ;; frame-persp-order ensures backwards compatibility with the early
+             ;; pre-release format of persp-state files: the frame may be just a
+             ;; hash table (old version), or it may be an instance of
+             ;; persp--state-frame (released version). The special case can be
+             ;; removed in the future, as there should be very few or no files
+             ;; left in the old format.
+             (lexical-let* ((frame-persp-table (if (hash-table-p frame)
+                                                   frame
+                                                 (persp--state-frame-persps frame)))
+                            (frame-persp-order (if (hash-table-p frame)
+                                                   (hash-table-keys frame)
+                                                 (reverse (persp--state-frame-order frame)))))
+               ;; iterate over the perspectives in the frame in the appropriate order
+               (cl-loop for persp in frame-persp-order do
+                        (lexical-let ((state-single (gethash persp frame-persp-table)))
+                          (persp-switch persp)
+                          (cl-loop for buffer in (persp--state-single-buffers state-single) do
+                                   (persp-add-buffer buffer))
+                          ;; XXX: split-window-horizontally is necessary for
```

Comment:
Please use "TODO" or "FIX" or something semantic.

Reply:
XXX _is_ semantic. I use it to mean "misleading code which pretty much has to be that way for external reasons". For reference, I use FIXME to mean "must be fixed before shipping, and ideally ASAP", and TODO to mean "not ideal, but can ship as is (and therefore probably will never get fixed)".

I note that the [Wikipedia article on comments](https://en.wikipedia.org/wiki/Comment_%28computer_programming%29#Tags) has a similar explanation, so it's not just me. 🙃